src/functionalTest/java/uk/gov/hmcts/reform/wacaseeventhandler/controllers/CaseEventHandlerTestingControllerFunctionalTest.java

```diff
@@ -93,26 +95,38 @@ public void given_post_event_using_test_rest_endpoints_should_create_task() {
             .extract()
             .path("[0].id");
 
-        Response response = findTaskDetailsForGivenTaskId(taskId);
-        String idempotencyKey = idempotencyKeyGenerator.generateIdempotencyKey(eventInformation.getEventInstanceId(),
-                                                                               "followUpNonStandardDirection");
-
-        response.then().assertThat()
-            .statusCode(HttpStatus.OK.value())
-            .and().contentType(MediaType.APPLICATION_JSON_VALUE)
-            .body("caseTypeId.value", equalToIgnoringCase("asylum"))
-            .body("jurisdiction.value", equalToIgnoringCase("ia"))
-            .body("idempotencyKey.value", is(idempotencyKey))
-            .body("dueDate.value", notNullValue())
-            .body("taskState.value", equalToIgnoringCase("unassigned"))
-            .body("hasWarnings.value", is(false))
-            .body("caseId.value", is(caseIdForTask))
-            .body("name.value", equalToIgnoringCase("Follow-up non-standard direction"))
-            .body("workingDaysAllowed.value", is(2))
-            .body("isDuplicate.value", is(false))
-            .body("delayUntil.value", CoreMatchers.notNullValue())
-            .body("taskId.value", equalToIgnoringCase("followUpNonStandardDirection"))
-            .body("warningList.value", is("[]"));
+        await().ignoreException(AssertionError.class)
+            .pollInterval(2, SECONDS)
+            .atMost(180, SECONDS)
+            .until(
+                () -> {
+                    Response response = findTaskDetailsForGivenTaskId(taskId);
+                    if (response != null) {
+                        String idempotencyKey = idempotencyKeyGenerator.generateIdempotencyKey(
+                            eventInformation.getEventInstanceId(),
+                            "followUpNonStandardDirection"
+                        );
+                        response.then().assertThat()
+                            .statusCode(HttpStatus.OK.value())
+                            .and().contentType(MediaType.APPLICATION_JSON_VALUE)
+                            .body("caseTypeId.value", equalToIgnoringCase("asylum"))
+                            .body("jurisdiction.value", equalToIgnoringCase("ia"))
+                            .body("idempotencyKey.value", is(idempotencyKey))
+                            .body("dueDate.value", notNullValue())
+                            .body("taskState.value", equalToIgnoringCase("unassigned"))
```

Comment:
The description of this PR is 

`Added unconfigured to taskState to CaseEventHandlerTestingControllerFunctionalTest.java`

but here the taskState is always `unassigned` ?

Reply:
At the time we were testing and we were having results within jenkins of unconfigured task but we got confirmation from Ravi: Unassigned means the Task has been created fully.  Unconfigured means it has not. Logic had to be changed.